var/spack/repos/builtin/packages/intel-oneapi-compilers/package.py

```diff
@@ -5,82 +5,77 @@
 
+    if platform == 'linux':
+        version('2021.1.2',
+                sha256='68d6cb638091990e578e358131c859f3bbbbfbf975c581fd0b4b4d36476d6f0a',
+                url='https://registrationcenter-download.intel.com/akdlm/irc_nas/17513/l_dpcpp-cpp-compiler_p_2021.1.2.63_offline.sh',
+                expand=False)
+        resource(name='fortran-installer',
+                 url='https://registrationcenter-download.intel.com/akdlm/irc_nas/17508/l_fortran-compiler_p_2021.1.2.62_offline.sh',
+                 sha256='29345145268d08a59fa7eb6e58c7522768466dd98f6d9754540d1a0803596829',
+                 expand=False,
+                 placement='fortran-installer',
+                 when='@2021.1.2')
+
     def __init__(self, spec):
-        self.component_info(
-            dir_name='compiler',
-            components=('intel.oneapi.lin.dpcpp-cpp-compiler-pro'
-                        ':intel.oneapi.lin.ifort-compiler'),
-            releases=releases,
-            url_name='HPCKit')
+        self.component_info(dir_name='compiler')
         super(IntelOneapiCompilers, self).__init__(spec)
 
-    def _join_prefix(self, path):
-        return join_path(self.prefix, 'compiler', 'latest', 'linux', path)
+    def _join_prefix(self, p):
+        return path.join(self.prefix, 'compiler', 'latest', 'linux', p)
 
     def _ld_library_path(self):
         dirs = ['lib',
-                'lib/x64',
-                'lib/emu',
-                'lib/oclfpga/host/linux64/lib',
-                'lib/oclfpga/linux64/lib',
-                'compiler/lib/intel64_lin',
-                'compiler/lib']
+                path.join('lib', 'x64'),
+                path.join('lib', 'emu'),
+                path.join('lib', 'oclfpga', 'host', 'linux64', 'lib'),
+                path.join('lib', 'oclfpga', 'linux64', 'lib'),
+                path.join('compiler', 'lib', 'intel64_lin'),
+                path.join('compiler', 'lib')]
         for dir in dirs:
             yield self._join_prefix(dir)
 
     def install(self, spec, prefix):
-        # For quick turnaround debugging, comment out line below and
-        # use the copy instead
+        # install cpp
+        # Copy instead of install to speed up debugging
+        # subprocess.run(f'cp -r /opt/intel/oneapi/compiler {prefix}', shell=True)
         super(IntelOneapiCompilers, self).install(spec, prefix)
-        # Copy installed compiler instead of running the installer
-        # from shutil import copytree
-        # copytree('/opt/intel/oneapi/compiler', join_path(prefix, 'compiler'),
-        #         symlinks=True)
 
+        # install fortran
+        super(IntelOneapiCompilers, self).install(
+            spec,
+            prefix,
+            installer_path=glob.glob(path.join('fortran-installer', '*'))[0])
+
+        # Some installers have a bug and do not return an error code when failing
+        if not path.isfile(path.join(prefix, 'compiler', 'latest', 'linux', 'bin', 'intel64', 'ifort')):
```

Comment:
According to the style checks (https://github.com/spack/spack/pull/22202/checks?check_run_id=2132375022) this line is too long and you'll need to break it up into multiple lines.

Reply:
Fixed all ```spack style``` issues.